File: 237 Delete Node in a Linked List/deleteNode.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def deleteNode(self, node):
        """
        :type node: ListNode
        :rtype: void Do not return anything, modify node in-place instead.
        """
        if node == None:
            return
        if node.next == None:
            node.next = None
            return
        
        node.val = node.next.val
        delNode = node.next
        node.next = delNode.next
        del delNode
        return

File: 237 Delete Node in a Linked List/deleteNode.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public void deleteNode(ListNode node) {
        
        if( node == null )
            return;
        if( node.next == null ){
            node = null;
            return;
        }
        node.val = node.next.val;
        ListNode delNode = node.next;
        node.next = delNode.next;
        delNode.next = null;
        return;
    }
}